ansible_base/rbac/evaluations.py

```diff
@@ -0,0 +1,91 @@
+from django.conf import settings
+
+from ansible_base.rbac import permission_registry
+from ansible_base.rbac.models import DABPermission, RoleDefinition, get_evaluation_model
+from ansible_base.rbac.validators import validate_codename_for_model
+
+"""
+RoleEvaluation or RoleEvaluationUUID models are the authority for permission evaluations,
+meaning, determining whether a user has a permission to an object.
+
+Methods needed for producing querysets (of objects a user has a permission to
+or users that have a permission to an object) or making single evaluations
+are defined on the RoleEvaluation model.
+
+This module has logic to attach those evaluation methods to the external
+models in an app using these RBAC internals.
+"""
+
+
+def has_super_permission(user, full_codename=None) -> bool:
+    "Analog to has_obj_perm but only evaluates to True if user has this permission system-wide"
+    if user._meta.model_name == permission_registry.user_model._meta.model_name:
+        # Super permission flags only exist for users, teams can use global roles
+        for super_flag in settings.ANSIBLE_BASE_BYPASS_SUPERUSER_FLAGS:
+            if getattr(user, super_flag):
+                return True  # User has admin flag like is_superuser
+        if full_codename:
+            for action, super_flag in settings.ANSIBLE_BASE_BYPASS_ACTION_FLAGS.items():
+                if full_codename == action and getattr(user, super_flag):
+                    return True  # User has action-specific flag like is_system_auditor
+    elif user._meta.model_name != permission_registry.team_model._meta.model_name:
+        raise RuntimeError(f'Evaluation methods are for users or teams, got {user._meta.model_name}: {user}')
+
+    if full_codename:
+        if full_codename in user.singleton_permissions():
+            return True  # User has system role for this action
+    return False
+
+
+def bound_singleton_permissions(self):
+    "Method attached to User model as singleton_permissions"
+    if not hasattr(self, '_singleton_permissions') or bound_singleton_permissions._team_clear_signal:
+        # values_list will make the return type set[str]
+        permission_qs = DABPermission.objects.values_list('codename', flat=True)
+        self._singleton_permissions = RoleDefinition.user_global_permissions(self, permission_qs=permission_qs)
+        bound_singleton_permissions._team_clear_signal = False
```

Comment:
Alright, this sucks. I just didn't want to do nothing here. I was _barely_ convinced that I should allow referencing team global permissions in the first place (you should only be evaluating user permissions), but there was some god-awful special case from AWX that drove it. The only time _this_ should matter would be if you were writing scripts to do something. It's not going to be airtight, but I think the case where it would matter is pretty absurdly niche.